File: tests/fixtures/v2-redirect-flow/positive/redirect_flow.go
package redirectflow

func direct(w ResponseWriter, r *Request) {
	http.Redirect(w, r, r.FormValue("direct"), 302)
}

func propagated(w ResponseWriter, r *Request) {
	requested := r.FormValue("propagated")
	alias := requested
	http.Redirect(w, r, alias, 302)
}

func parsed(w ResponseWriter, r *Request) {
	destination, _ := url.Parse(r.FormValue("parsed"))
	http.Redirect(w, r, destination.String(), 302)
}

func validLocal(destination *url.URL) bool {
	return !destination.IsAbs()
}

File: tests/fixtures/v2-redirect-flow/positive/RedirectFlow.cs
class RedirectFlow
{
    void Direct(dynamic Request, dynamic Response)
    {
        Response.Redirect(Request.Query["direct"]);
    }

    void Propagated(dynamic Request, dynamic Response)
    {
        var requested = Request.Query["propagated"];
        var forwardedValue = requested;
        Response.Redirect(forwardedValue);
    }

    void Parsed(dynamic Request, dynamic Response)
    {
        var destination = new Uri(Request.Query["parsed"]);
        Response.Redirect(destination.ToString());
    }

    bool ValidLocal(string destination)
    {
        return Url.IsLocalUrl(destination);
    }
}
